src/systemctl/systemctl-edit.c

```diff
@@ -483,6 +483,18 @@ static int trim_edit_markers(const char *path) {
                 strshorten(contents_start, contents_end - contents_start);
 
         contents_start = strstrip(contents_start);
+        if (*contents_start && !endswith(contents_start, "\n")) {
```

Comment:
according to our coding style, we don't want to use C's downgrade-to-bool feature unless the type of the variable is actually boolean or boolean-like – or is a pointer. but *contents_start here is a char, i.e. neither boolean nor pointer. Hence please write:

```c
if (contents_start[0] != 0 && … 
```

Reply:
so the second part of the expression is redundant. strstrip() removes trailing whitespace anyway, which means newlines are removed.